MediaBrowser.Providers/TV/TheTVDB/TvdbSeasonImageProvider.cs

```diff
@@ -62,91 +57,78 @@ public async Task<IEnumerable<RemoteImageInfo>> GetImages(BaseItem item, Cancell
+            foreach (Image image in images)
             {
-                using (var streamReader = new StreamReader(fileStream, Encoding.UTF8))
+                var imageInfo = new RemoteImageInfo
                 {
-                    // Use XmlReader for best performance
-                    using (var reader = XmlReader.Create(streamReader, settings))
-                    {
-                        reader.MoveToContent();
-                        reader.Read();
+                    RatingType = RatingType.Score,
+                    CommunityRating = (double?)image.RatingsInfo.Average,
+                    VoteCount = image.RatingsInfo.Count,
+                    Url = TVUtils.BannerUrl + image.FileName,
+                    ProviderName = ProviderName,
+                    Language = languages.FirstOrDefault(lang => lang.Id == image.LanguageId)?.Abbreviation,
+                    ThumbnailUrl = TVUtils.BannerUrl + image.Thumbnail
+                };
 
-                        // Loop through each element
-                        while (!reader.EOF && reader.ReadState == ReadState.Interactive)
-                        {
-                            cancellationToken.ThrowIfCancellationRequested();
+                var resolution = image.Resolution.Split('x');
+                if (resolution.Length == 2)
+                {
+                    imageInfo.Width = Convert.ToInt32(resolution[0]);
+                    imageInfo.Height = Convert.ToInt32(resolution[1]);
+                }
 
-                            if (reader.NodeType == XmlNodeType.Element)
-                            {
-                                switch (reader.Name)
-                                {
-                                    case "Banner":
-                                        {
-                                            if (reader.IsEmptyElement)
-                                            {
-                                                reader.Read();
-                                                continue;
-                                            }
-                                            using (var subtree = reader.ReadSubtree())
-                                            {
-                                                AddImage(subtree, list, seasonNumber);
-                                            }
-                                            break;
-                                        }
-                                    default:
-                                        reader.Skip();
-                                        break;
-                                }
-                            }
-                            else
-                            {
-                                reader.Read();
-                            }
-                        }
-                    }
+                if (string.Equals(image.KeyType, "season", StringComparison.OrdinalIgnoreCase))
```

Comment:
_repeating so those issues won't get lost_

cannot we do it using some `switch`?

Reply:
When microbenching yesterday, the switch + lower will be slower.

Reply:
This scanning happens once in a few hours at best, so a few extra nanoseconds won't hurt, but readability improves with `switch` IMHO.

Compare readability:
_current_:
```cs
                if (string.Equals(image.KeyType, "season", StringComparison.OrdinalIgnoreCase))
                {
                    imageInfo.Type = ImageType.Primary;
                }
                else if (string.Equals(image.KeyType, "seasonwide", StringComparison.OrdinalIgnoreCase))
                {
                    imageInfo.Type = ImageType.Banner;
                }
                else if (string.Equals(image.KeyType, "fanart", StringComparison.OrdinalIgnoreCase))
                {
                    imageInfo.Type = ImageType.Backdrop;
                }
```
_proposed_:
```cs
                switch (image.KeyType.ToLowerInvariant()) {
                    case "season": imageInfo.Type = ImageType.Primary;
                    case "seasonwide": imageInfo.Type = ImageType.Banner;
                    case "fanart": imageInfo.Type = ImageType.Backdrop;
                }
```

Reply:
Readability-wise it isn't that different

```cs
                switch (image.KeyType.ToLowerInvariant())
                {
                    case "season":
                        imageInfo.Type = ImageType.Primary;
                        break;
                    case "seasonwide":
                        imageInfo.Type = ImageType.Banner;
                        break;
                    case "fanart":
                        imageInfo.Type = ImageType.Backdrop;
                        break;
                }
```